test/tools/yulopti.cpp

```diff
@@ -214,47 +361,48 @@ class YulOpti
 					case '#':
 						return;
 					case ',':
-						VarNameCleaner::run(m_context, *m_ast);
+						runVarNameCleaner();
 						// VarNameCleaner destroys the unique names guarantee of the disambiguator.
 						disambiguated = false;
 						break;
 					case ';':
-					{
-						Object obj;
-						obj.code = m_ast;
-						StackCompressor::run(m_dialect, obj, true, 16);
+						runStackCompressor();
 						break;
-					}
 					default:
-						OptimiserSuite{m_context}.runSequence(
-							std::string_view(&option, 1),
-							*m_ast
-						);
+						runSequence(string_view(&option, 1));
 				}
-				_source = AsmPrinter{m_dialect}(*m_ast);
+
+				resetNameDispenser();
+				runCodeAnalyzer(errorReporter);
 			}
 			catch (...)
 			{
 				cerr << endl << "Exception during optimiser step:" << endl;
 				cerr << boost::current_exception_diagnostic_information() << endl;
 			}
 			cout << "----------------------" << endl;
-			cout << _source << endl;
+			printObject();
```

Comment:
@cameel I think this is more a design question about the behavior of the tool. Now that it can read and work with nested objects, should their code also be printed?

Reply:
Yes. I haven't reviewed the changes here but I'm assuming that now the tool can work both with a Yul block (as until now) and with a full object. In that case, with an object, the optimizer is also processing subobjects and I'd expect to see them in the output. Stripping them may give you an object that does not even compile (e.g. when you have `datacopy()` that refers to a subobject). If the user does not want to see subobjects, he can pass in a block or remove them from the input.

We may also want to add an option to ignore nested objects but I wouldn't add it until it turns out that this is something that's actually useful in practice.

Reply:
@dflupu, please consider this.

Reply:
This was how I had intended it to work, but it often didn't because I didn't properly initialize `m_inputWasCodeBlock` to false. Please try running the tool again and let me know if it is printing nested objects as expected now.

Reply:
Ok, I confirmed that it now prints subobjects as well.